examples/fabric/byot/trainer.py

```diff
@@ -0,0 +1,409 @@
+
+        iterable = self.pbar_wrapper(
+            val_loader,
+            total=min(len(val_loader), limit_batches),
+            desc="Validation Sanity Check" if self.global_step == 0 else "Validation",
+        )
+
+        for batch_idx, batch in enumerate(iterable):
+
+            # end epoch if stopping training completely or max batches for this epoch reached
+            if self.should_stop or batch_idx >= limit_batches:
+                self.fabric.call("on_validation_epoch_end")
+                return
+
+            self.fabric.call("on_validation_batch_start", batch, batch_idx)
+
+            out = model.validation_step(batch, batch_idx)
+            # avoid gradients in stored/accumulated values -> prevents potential OOM
+            out = apply_to_collection(out, torch.Tensor, lambda x: x.detach())
+
+            self.fabric.call("on_validation_batch_end", out, batch, batch_idx)
+            self._current_val_return = out
+
+            self._format_iterable(iterable, self._current_val_return, "val")
+
+        self.fabric.call("on_validation_epoch_end")
+
+        self.fabric.call("on_validation_model_train")
+        # TODO: Inference mode?
+        torch.set_grad_enabled(True)
+
+    def training_step(self, model, optimizer, batch, batch_idx):
+        outputs: Union[torch.Tensor, Mapping[str, Any]] = model.training_step(batch, batch_idx=batch_idx)
+
+        loss = outputs if isinstance(outputs, torch.Tensor) else outputs["loss"]
+
+        self.fabric.call("on_before_backward", loss)
+        self.fabric.backward(loss)
+        self.fabric.call("on_after_backward")
+        # TODO: reroute configure_gradient_clipping in LM to fabric.gradient clipping
+        # self.fabric.call('configure_gradient_clipping')
+
+        # avoid gradients in stored/accumulated values -> prevents potential OOM
+        self._current_train_return = apply_to_collection(outputs, dtype=torch.Tensor, function=lambda x: x.detach())
+
+        return loss
+
+    def step_scheduler(
+        self,
+        model: LightningModule,
+        scheduler_cfg: Optional[Mapping[str, Union[LRScheduler, bool, str, int]]],
+        level: Literal["step", "epoch"],
+        current_value: int,
+    ):
+        # no scheduler
+        if scheduler_cfg is None:
+            return
+
+        # wrong interval (step vs. epoch)
+        if scheduler_cfg["interval"] != level:
+            return
+
+        # right interval, but wrong step wrt frequency
+        if current_value % cast(int, scheduler_cfg["frequency"]) != 0:
+            return
+
+        # assemble potential monitored values
+        possible_monitor_vals = {None: None}
+        if isinstance(self._current_train_return, torch.Tensor):
+            possible_monitor_vals.update("train_loss", self._current_train_return)
+        elif isinstance(self._current_train_return, Mapping):
+            possible_monitor_vals.update({"train_" + k: v for k, v in self._current_train_return.items()})
```

Comment:
In you example, this would get prefixed twice, because the training step returns `train_accuracy` already

Reply:
Nope, that's not true. In my example, the variable is called `train_accuracy` as I can't just use `accuracy` to avoid shadowing the import from TM, but the key this is returned with is just `accuracy`.